protocols/smart-transfer-protocol.openapi.yaml

```diff
@@ -0,0 +1,228 @@
+openapi: 3.0.3
+
+info:
+  title: ODF Smart Transfer Protocol (HTTP endpoints part)
+  description:
+    Open Data Fabric (ODF) extended protocol for dataset exchange between remote parties.
+    Defines REST API endpoints that enable simple operation and upgrading to asynchronous API protocol.
+  license:
+    name: Apache 2.0
+    url: http://www.apache.org/licenses/LICENSE-2.0.html
+  version: 0.1.0
+
+components:
+
+  parameters:
+
+    Reference:
+      in: path
+      name: reference
+      description: Name of the reference
+      example: head
+      required: true
+      schema:
+        type: string
+        enum: [head]
+
+    BlockHash:
```

Comment:
I don't see a good reason to differentiate block hashes from hashes of other objects (data/checkpoint files) - they will always have the same representation and an algorithm, so I would use `ObjectPhysicalHash` for blocks endpoint too.

Reply:
I renamed the parameter definitions a bit, made them lowercase, so that they are not confused with types.